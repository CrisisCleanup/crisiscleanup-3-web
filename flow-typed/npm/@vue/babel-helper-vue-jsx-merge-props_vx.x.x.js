// flow-typed signature: d7a23c53f4091ef2bf1df8afa5eb43c4
// flow-typed version: <<STUB>>/@vue/babel-helper-vue-jsx-merge-props_v^1.0.0/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-helper-vue-jsx-merge-props'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-helper-vue-jsx-merge-props' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper.js' {
  declare module.exports: $Exports<'@vue/babel-helper-vue-jsx-merge-props/dist/helper'>;
}
