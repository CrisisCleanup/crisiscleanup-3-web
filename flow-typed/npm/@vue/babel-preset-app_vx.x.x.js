// flow-typed signature: 6ba91a92845d9a556a5bcdcc4f033fed
// flow-typed version: <<STUB>>/@vue/babel-preset-app_v^4.3.1/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-app'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-app' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-app/polyfillsPlugin' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-app/index' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/index.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/polyfillsPlugin.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app/polyfillsPlugin'>;
}
