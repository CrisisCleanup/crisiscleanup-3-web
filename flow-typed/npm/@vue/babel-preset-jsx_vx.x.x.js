// flow-typed signature: eeecaeff29e4a3a6c63c4fe7e16a0eb3
// flow-typed version: <<STUB>>/@vue/babel-preset-jsx_v^1.1.2/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-jsx'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-jsx' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-jsx/dist/plugin.cjs' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-jsx/dist/plugin.cjs.js' {
  declare module.exports: $Exports<'@vue/babel-preset-jsx/dist/plugin.cjs'>;
}
