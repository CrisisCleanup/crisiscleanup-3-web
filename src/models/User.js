import { AuthService } from '@/services/auth.service';
import Language from '@/models/Language';
import Role from '@/models/Role';
import moment from 'moment';
import CCUModel from '@/models/model';

export default class User extends CCUModel {
  static entity = 'users';

  static fields() {
    return {
      id: this.attr(),
      first_name: this.string(''),
      last_name: this.string(''),
      email: this.string(''),
      mobile: this.string(''),
      roles: this.attr(null),
      active_roles: this.attr(null),
      files: this.attr(null),
      organization: this.attr(null),
      states: this.attr({}),
      preferences: this.attr({}),
      permissions: this.attr({}),
      beta_features: this.attr({}),
      primary_language: this.attr(null),
      secondary_language: this.attr(null),
      accepted_terms_timestamp: this.attr(null),
      accepted_terms: this.attr(null),
      social: this.attr({}),
      referring_user: this.attr({}),
      lineage: this.attr([]),
    };
  }

  get profilePictureUrl() {
    if (this.files && this.files.length) {
      const profilePictures = this.files.filter(
        (file) => file.file_type_t === 'fileTypes.user_profile_picture',
      );
      if (profilePictures.length) {
        return profilePictures[0].url;
      }
    }
    return `https://api.adorable.io/avatars/285/ccu-user-${this.id}.png`;
  }

  get currentRole() {
    return Role.query().whereIdIn(this.active_roles).get()[0];
  }

  get referringUser() {
    return User.find(this.referring_user);
  }

  get languages() {
    const languageList = [];
    if (this.primary_language) {
      languageList.push(Language.find(this.primary_language));
    }
    if (this.secondary_language) {
      languageList.push(Language.find(this.secondary_language));
    }
    return languageList;
  }

  get notificationSettings() {
    const settings = {
      has_notifications: false,
    };
    if (this.preferences && this.preferences.notification_settings) {
      return this.preferences.notification_settings;
    }
    return settings;
  }

  get full_name() {
    return `${this.first_name} ${this.last_name}`;
  }

  get facebook() {
    return this.social && this.social.facebook;
  }

  get twitter() {
    return this.social && this.social.twitter;
  }

  get isAdmin() {
    return this.active_roles.includes(1);
  }

  getStatesForIncident(incidentId, fallback = true) {
    if (
      this.states &&
      this.states.incidents &&
      this.states.incidents[incidentId]
    ) {
      return this.states.incidents[incidentId];
    }
    if (fallback) {
      return this.states;
    }
    return null;
  }

  static apiConfig = {
    actions: {
      login(email, password) {
        return this.post(
          `/api-token-auth`,
          {
            email,
            password,
          },
          { save: false },
        );
      },
      inviteUser(email, organization = null) {
        const data = {
          invitee_email: email,
        };
        if (organization) {
          data.organization = organization;
        }
        return this.post(`/invitations`, data, { save: false });
      },
      acceptInvite({ token, first_name, last_name, password, mobile, title }) {
        return this.post(
          `/invitations/accept`,
          {
            invitation_token: token,
            first_name,
            last_name,
            password,
            mobile,
            title,
          },
          { save: false },
        );
      },
      orphan(id) {
        return this.patch(`/users/${id}/orphan`);
      },
      addFile(id, file, type) {
        return this.post(
          `/users/${id}/files`,
          {
            file,
            type_t: type,
          },
          { save: false },
        );
      },
      deleteFile(id, file) {
        return this.delete(
          `/users/${id}/files`,
          {
            data: { file },
          },
          { save: false },
        );
      },
      async updateUserState(globalStates, incidentStates, reload = false) {
        /* Update user states JSON with new states.

           To be backwards-compatible with clients without per-incident states,
           update top-level with both globalStates and incidentStates
           and then update state for current incident with incidentStates.
        */
        let currentUser = User.find(AuthService.getUser().user_claims.id);
        const states = currentUser.states || {};
        const currentIncident = states.incident;
        let updatedStates = {
          ...states,
          ...globalStates,
          ...incidentStates,
        };
        let updatedIncidentStates = states.incidents || {};
        if (incidentStates) {
          updatedIncidentStates = {
            ...updatedIncidentStates,
            [currentIncident]: incidentStates,
          };
        }
        updatedStates = {
          ...updatedStates,
          ...{ incidents: updatedIncidentStates },
        };
        await User.update({
          where: currentUser.id,
          data: {
            states: updatedStates,
          },
        });
        currentUser = User.find(AuthService.getUser().user_claims.id);
        await this.patch(
          `/users/${currentUser.id}`,
          {
            states,
          },
          { save: false },
        );
        if (reload) {
          await this.get('/users/me', {});
        }
      },
      async updateUserPreferences(preferences, reload = false) {
        const currentUser = User.find(AuthService.getUser().user_claims.id);
        const newPreferences = {
          ...currentUser.preferences,
          ...preferences,
        };
        await this.patch(
          `/users/${currentUser.id}`,
          {
            preferences: newPreferences,
          },
          { save: false },
        );
        if (reload) {
          await this.get('/users/me', {});
        }
      },
      async acceptTerms() {
        const currentUser = User.find(AuthService.getUser().user_claims.id);
        await this.patch(`/users/${currentUser.id}`, {
          accepted_terms: true,
          accepted_terms_timestamp: moment().toISOString(),
        });
      },
    },
  };
}
