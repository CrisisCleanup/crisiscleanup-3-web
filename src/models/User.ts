import moment from 'moment';
import Bowser from 'bowser';
import * as Sentry from '@sentry/browser';
import { useRouter } from 'vue-router';
import { Model } from '@vuex-orm/core';
import type { Config } from '@vuex-orm/plugin-axios';
import { AuthService } from '../services/auth.service';
import Language from './Language';
import Role from './Role';
import CCUModel from '@/models/model';

export default class User extends CCUModel<User> {
  static entity = 'users';

  id!: string;

  primary_language!: string;

  secondary_language!: string;

  organization!: any;

  active_roles!: any[];

  referring_user!: string;

  preferences!: Record<string, any>;

  first_name!: string;

  last_name!: string;

  email!: string;

  mobile!: string;

  accepted_terms_timestamp!: string;

  states!: Record<string, any>;

  social!: Record<string, any>;

  files!: any[];

  lineage!: any[];

  permissions!: Record<string, boolean>;

  beta_features!: string[];

  static fields() {
    return {
      id: this.attr(''),
      first_name: this.string(''),
      last_name: this.string(''),
      email: this.string(''),
      mobile: this.string(''),
      roles: this.attr(null),
      active_roles: this.attr(null),
      files: this.attr(null),
      organization: this.attr(null),
      states: this.attr({}),
      preferences: this.attr({}),
      permissions: this.attr({}),
      beta_features: this.attr({}),
      primary_language: this.attr(null),
      secondary_language: this.attr(null),
      accepted_terms_timestamp: this.attr(null),
      accepted_terms: this.attr(null),
      social: this.attr({}),
      referring_user: this.attr({}),
      lineage: this.attr([]),
    };
  }

  static afterUpdate(model: User) {
    if (model.id === User.store().getters['auth/userId']) {
      AuthService.updateUser(model.$toJson());
      Sentry.setUser(model.$toJson());
      Sentry.setContext('user_states', model.states);
      Sentry.setContext('user_preferences', model.preferences);
      // User.store().commit("auth/setAcl", useRouter());
    }
  }

  get hasProfilePicture() {
    if (this.files && this.files.length > 0) {
      const profilePictures = this.files.filter(
        (file) => file.file_type_t === 'fileTypes.user_profile_picture',
      );
      return profilePictures.length;
    }

    return false;
  }

  get profilePictureUrl() {
    if (this.files && this.files.length > 0) {
      const profilePictures = this.files.filter(
        (file) => file.file_type_t === 'fileTypes.user_profile_picture',
      );
      if (profilePictures.length > 0) {
        return profilePictures[0].large_thumbnail_url;
      }
    }

    return `https://avatars.dicebear.com/api/bottts/${this.full_name}.svg`;
  }

  get currentRole() {
    return Role.query().whereIdIn(this.active_roles).get()[0];
  }

  get referringUser() {
    return User.find(this.referring_user);
  }

  get languages() {
    const languageList: any[] = [];
    if (this.primary_language) {
      languageList.push(Language.find(this.primary_language));
    }

    if (this.secondary_language) {
      languageList.push(Language.find(this.secondary_language));
    }

    return languageList;
  }

  get languageIds() {
    const languageList: any[] = [];
    if (this.primary_language) {
      languageList.push(this.primary_language);
    }

    if (this.secondary_language) {
      languageList.push(this.secondary_language);
    }

    return languageList;
  }

  get notificationSettings() {
    const settings = {
      has_notifications: false,
    };
    if (this.preferences && this.preferences.notification_settings) {
      return this.preferences.notification_settings;
    }

    return settings;
  }

  get full_name() {
    return `${this.first_name} ${this.last_name}`;
  }

  get facebook() {
    return this.social && this.social.facebook;
  }

  get twitter() {
    return this.social && this.social.twitter;
  }

  get isAdmin() {
    return this.active_roles.includes(1);
  }

  get isPrimaryContact() {
    return this.active_roles.includes(3);
  }

  getStatesForIncident(incidentId: string, fallback = true) {
    if (
      this.states &&
      this.states.incidents &&
      this.states.incidents[incidentId]
    ) {
      return this.states.incidents[incidentId];
    }

    if (fallback) {
      return this.states;
    }

    return null;
  }

  static apiConfig: Config = {
    actions: {
      login(email: string, password: string) {
        return this.post(
          `/api-token-auth`,
          {
            email,
            password,
          },
          { save: false },
        );
      },
      inviteUser(email: string, organization = null) {
        const data: Record<string, any> = {
          invitee_email: email,
        };
        if (organization) {
          data.organization = organization;
        }

        return this.post(`/invitations`, data, { save: false });
      },
      acceptInvite({
        token,
        first_name,
        last_name,
        password,
        mobile,
        title,
      }: Record<string, any>) {
        return this.post(
          `/invitations/accept`,
          {
            invitation_token: token,
            first_name,
            last_name,
            password,
            mobile,
            title,
          },
          { save: false },
        );
      },
      orphan(id: string) {
        return this.patch(`/users/${id}/orphan`);
      },
      addFile(id: string, file: string, type: string) {
        return this.post(
          `/users/${id}/files`,
          {
            file,
            type_t: type,
          },
          { save: false },
        );
      },
      deleteFile(id: string, file: string) {
        return this.delete(
          `/users/${id}/files`,
          {
            data: { file },
          },
          { save: false },
        );
      },
      async clearUserStates() {
        const currentUser = User.find(User.store().getters['auth/userId']);
        await this.patch(
          `/users/${currentUser?.id}`,
          {
            states: {},
          },
          { save: false },
        );
        await this.get('/users/me');
      },
      async clearUserPreferences() {
        const currentUser = User.find(User.store().getters['auth/userId']);
        await this.patch(
          `/users/${currentUser?.id}`,
          {
            preferences: {},
          },
          { save: false },
        );
        await this.get('/users/me');
      },
      async updateUserState(
        globalStates: Record<string, any>,
        incidentStates: Record<string, any>,
        reload = false,
      ) {
        /* Update user states JSON with new states.

           To be backwards-compatible with clients without per-incident states,
           update top-level with both globalStates and incidentStates
           and then update state for current incident with incidentStates.
        */
        let currentUser = User.find(User.store().getters['auth/userId']);

        if (!currentUser) {
          return;
        }

        const states = (currentUser && currentUser.states) || {};
        const currentIncident = states.incident;
        let updatedStates = {
          ...states,
          ...globalStates,
          ...incidentStates,
        };
        let updatedIncidentStates = states.incidents || {};
        if (incidentStates) {
          updatedIncidentStates = {
            ...updatedIncidentStates,
            [currentIncident]: incidentStates,
          };
        }

        updatedStates = {
          ...updatedStates,
          incidents: updatedIncidentStates,
          userAgent: Bowser.parse(window.navigator.userAgent),
        };
        await User.update({
          where: currentUser.id,
          data: {
            states: updatedStates,
          },
        });
        currentUser = User.find(AuthService.getUser().user_claims.id);
        await this.patch(
          `/users/${currentUser?.id}`,
          {
            states: updatedStates,
          },
          { save: false },
        );
        if (reload) {
          await this.get('/users/me');
        }
      },
      async updateUserPreferences(
        preferences: Record<string, any>,
        reload = false,
      ) {
        const currentUser = User.find(AuthService.getUser().user_claims.id);
        if (!currentUser) {
          return;
        }

        const newPreferences = {
          ...currentUser.preferences,
          ...preferences,
        };
        await this.patch(
          `/users/${currentUser.id}`,
          {
            preferences: newPreferences,
          },
          { save: false },
        );
        if (reload) {
          await this.get('/users/me', {});
        }
      },
      async acceptTerms() {
        const currentUser = User.find(AuthService.getUser().user_claims.id);
        await this.patch(`/users/${currentUser?.id}`, {
          accepted_terms: true,
          accepted_terms_timestamp: moment().toISOString(),
        });
      },
    },
  };
}
