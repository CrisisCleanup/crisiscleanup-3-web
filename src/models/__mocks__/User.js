import { MockUsers } from "@/utils/testing";

export default {
  all: jest.fn(() => MockUsers),
  find: jest.fn(() => MockUsers[0]),
  database: jest.fn(() => MockUsers)
};
