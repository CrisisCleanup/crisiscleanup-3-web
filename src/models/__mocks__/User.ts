import { MockUsers, mockModel } from '@/utils/testing';

// @ts-ignore
export default mockModel(MockUsers, {
  api: {
    inviteUser: jest.fn(),
  },
});
