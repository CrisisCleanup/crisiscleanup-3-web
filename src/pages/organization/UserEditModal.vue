<template>
  <modal
    modal-classes="w-1/3"
    :title="$t('~~Edit User')"
    closeable
    @close="$emit('close')"
  >
    <div class="flex px-10 py-5">
      <div class="w-1/2 flex items-start justify-center">
        <img
          class="rounded-full profile-image mr-16 w-40"
          :src="user.profilePictureUrl"
          :alt="$t('userView.profile_picture')"
        />
      </div>
      <div class="w-1/2 flex flex-col">
        <base-input
          size="large"
          class="m-2"
          :value="user.first_name"
          :placeholder="$t('profileUser.first_name_placeholder')"
          required
          @input="
            value => {
              updateUser(value, 'first_name');
            }
          "
        />
        <base-input
          size="large"
          class="m-2"
          :value="user.last_name"
          :placeholder="$t('profileUser.last_name_placeholder')"
          required
          @input="
            value => {
              updateUser(value, 'last_name');
            }
          "
        />
        <base-input
          size="large"
          class="m-2"
          :value="user.mobile"
          :placeholder="$t('profileUser.mobile_placeholder')"
          required
          @input="
            value => {
              updateUser(value, 'mobile');
            }
          "
        />
        <base-input
          :value="user.email"
          size="large"
          class="m-2"
          :placeholder="$t('profileUser.email_placeholder ')"
          disabled
        />
        <form-select
          v-model="user.roles"
          class="flex-grow m-2 border border-crisiscleanup-dark-100"
          :value="user.roles"
          multiple
          :options="roles"
          item-key="id"
          label="name_t"
          size="large"
          select-classes="bg-white border text-xs profile-select p-1"
        />
      </div>
    </div>
    <div slot="footer" class="p-3 flex justify-end">
      <base-button
        :text="$t('actions.cancel')"
        class="ml-2 p-3 px-6 mr-1 text-xs border border-black"
        :action="
          () => {
            $emit('close');
          }
        "
      />
      <base-button
        variant="solid"
        :action="saveUser"
        :text="$t('actions.save')"
        class="ml-2 p-3 px-6 text-xs"
      />
    </div>
  </modal>
</template>

<script>
import User from '@/models/User';
import Role from '@/models/Role';
export default {
  name: 'UserEditModal',
  props: {
    user: {
      type: Object,
      default: () => {
        return {};
      },
    },
  },
  methods: {
    updateUser(value, key) {
      const { user } = this;
      User.update({
        where: user.id,
        data: {
          [key]: value,
        },
      });
    },
    saveUser() {
      this.$emit('save');
    },
  },
  computed: {
    roles() {
      return Role.all();
    },
    userRoles() {
      return Role.query()
        .whereIdIn(this.user.roles)
        .get();
    },
  },
};
</script>

<style scoped></style>
