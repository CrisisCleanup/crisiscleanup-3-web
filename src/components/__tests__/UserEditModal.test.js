/**
 *
 * Tests for WorkTypeRequestModal
 *
 * Components
 */

import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import User from '@/models/__mocks__/User';
import UserEditModal from '@/pages/organization/UserEditModal';

const localVue = createLocalVue();
localVue.use(Vuex);
jest.mock('@/models/User');
jest.mock('@/models/Role');

const mocks = {
  $t: (key) => key,
  $store: new Vuex.Store(),
};

const mountWithOptions = (props) =>
  shallowMount(UserEditModal, {
    stubs: {
      modal: true,
      'base-input': true,
      'form-select': true,
      'base-button': true,
    },
    propsData: {
      ...props,
    },
    store: mocks.$store,
    localVue,
    mocks,
  });

describe('UserEditModal', () => {
  it('should not log any errors', () => {
    const spy = jest.spyOn(global.console, 'error');
    mountWithOptions({});
    expect(spy).not.toHaveBeenCalled();
  });

  it('should emit save correctly', () => {
    const wrapper = mountWithOptions({
      user: User.find(1),
    });
    wrapper.vm.saveUser();
    expect(wrapper.emitted().save.length).toBe(1);
  });
});
