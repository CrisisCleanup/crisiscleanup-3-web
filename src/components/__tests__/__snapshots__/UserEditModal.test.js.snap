// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEditModal show modal with options 1`] = `
<modal-stub
  closeable=""
  modal-classes="w-1/3"
  title="~~Edit User"
>
  <div
    class="flex px-10 py-5"
  >
    <div
      class="w-1/2 flex items-start justify-center"
    >
      <img
        alt="userView.profile_picture"
        class="rounded-full profile-image mr-16 w-40"
      />
    </div>
     
    <div
      class="w-1/2 flex flex-col"
    >
      <input
        class="m-2"
        placeholder="profileUser.first_name_placeholder"
        required="required"
        size="large"
        value="First"
      />
       
      <input
        class="m-2"
        placeholder="profileUser.last_name_placeholder"
        required="required"
        size="large"
        value="Last"
      />
       
      <input
        class="m-2"
        placeholder="profileUser.mobile_placeholder"
        required="required"
        size="large"
      />
       
      <input
        class="m-2"
        disabled="disabled"
        placeholder="profileUser.email_placeholder "
        size="large"
      />
       
      <input
        class="flex-grow m-2 border border-crisiscleanup-dark-100"
        item-key="id"
        label="name_t"
        multiple="multiple"
        options="[object Object],[object Object]"
        select-classes="bg-white border text-xs profile-select p-1"
        size="large"
        type="select"
      />
    </div>
  </div>
   
  <div
    class="p-3 flex justify-end"
  >
    <button
      action="function () {
        _vm.$emit('close');
      }"
      class="ml-2 p-3 px-6 mr-1 text-xs border border-black"
      text="actions.cancel"
    />
     
    <button
      action="function () { [native code] }"
      class="ml-2 p-3 px-6 text-xs"
      text="actions.save"
      type="primary"
    />
  </div>
</modal-stub>
`;
